cirq-core/cirq/sim/simulator.py

```diff
@@ -876,6 +889,55 @@ def sample_measurement_ops(
         )
 
 
+# When removing this, also remove the check in simulate_sweep_iter.
+# Basically there should be no "final_step_result" anywhere in the project afterwards.
+def _deprecated_step_result_parameter(
```

Comment:
The intent is to remove this method after the next release, right?

Reply:
Yes, this was purely to reduce the redundancy of those deprecation attributes. It's not intended to be general purpose or used for anything else ever, and will go away when the deprecated parameters do.

Reply:
After this, #5270 will be a small PR, then ActOnArgs rename after that.